### DISCUSSION TITLE
Vehicle count in merge/roundabout env

### DISCUSSION THREAD

**Discussion by cclvr** [eleurent/highway-env#179] [General]
Hi @eleurent , I'm trying to use the merge/roundabout environment, and it seems that the vehicle count and the number of lanes cannot be set as the same way in highway env as follows:
        config.update({
            "observation": {
                "type": "Kinematics"
            },
            "lanes_count": 4,
            "vehicles_count": 50,
            "duration": 40,  # [s]
            "initial_spacing": 2,
            "collision_reward": self.COLLISION_REWARD
        })
So can they be changed in merge or roundablut? If yes, how?

By the way, the type built in for merge task and roundabout task is  OccupancyGrid, can I change it into Kinematic? Is it still valid?

Looking forward to your replay.

**Comment by eleurent** [comment]
Hi @cclvr 
No, the roundabout environment currently has a non-configurable number of vehicles and lanes.
I will make it configurable when I have a few minutes to spare, but meanwhile I encourage you to have a look at https://github.com/eleurent/highway-env/blob/master/highway_env/envs/merge_env.py#L65 and https://github.com/eleurent/highway-env/blob/master/highway_env/envs/merge_env.py#L97 , it should not be too difficult to modify.


Regarding the observation type, you can certainly change it to Kinematic. Be careful to change the features_range used for normalisation, since by default it was chosen (for highways) to a scale of about 100m for x and about 15m for y, which would not make sense in the roundabout scene.
